public/components/preview_panel/__tests__/preview_panel.test.tsx

```diff
@@ -83,4 +87,34 @@ describe('<PreviewPanel />', () => {
     setup({});
     await waitFor(() => expect(screen.getByText('Responding on 3 of 3 nodes')).toBeInTheDocument());
   });
+
+  it('should NOT render nodes during model profile API loading', async () => {
+    jest.useFakeTimers();
+    const getModelMock = jest.spyOn(APIProvider.getAPI('profile'), 'getModel').mockReturnValue(
+      new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({
+            id: 'model-1-id',
+            target_worker_nodes: ['node-1', 'node-2', 'node-3'],
+            worker_nodes: ['node-1'],
+            not_worker_nodes: ['node-2', 'node-3'],
+          });
+        }, 3000);
+      })
+    );
+    setup({});
+    expect(screen.queryByText('node-1')).not.toBeInTheDocument();
+    expect(screen.queryByText('node-2')).not.toBeInTheDocument();
+    expect(screen.queryByText('node-3')).not.toBeInTheDocument();
+
+    jest.advanceTimersByTime(3000);
+    await act(async () => {
+      await getModelMock.mock.results[0].value;
+    });
+
+    expect(await screen.getByText('node-1')).toBeInTheDocument();
+    expect(await screen.getByText('node-2')).toBeInTheDocument();
+    expect(await screen.getByText('node-3')).toBeInTheDocument();
```

Comment:
Shall we simply do `waitFor`?
```suggestion
    await waitFor(() => {
      expect(screen.getByText('node-1')).toBeInTheDocument();
      expect(screen.getByText('node-2')).toBeInTheDocument();
      expect(screen.getByText('node-3')).toBeInTheDocument();
    });
```